src/components/RoomNameInput/index.js

```diff
@@ -1,95 +1,68 @@
-import React, {Component} from 'react';
+import React, {forwardRef, useState} from 'react';
 import _ from 'underscore';
 import CONST from '../../CONST';
 import withLocalize from '../withLocalize';
 import TextInput from '../TextInput';
 import * as roomNameInputPropTypes from './roomNameInputPropTypes';
 import * as RoomNameInputUtils from '../../libs/RoomNameInputUtils';
 
-class RoomNameInput extends Component {
-    constructor(props) {
-        super(props);
-
-        this.setModifiedRoomName = this.setModifiedRoomName.bind(this);
-        this.setSelection = this.setSelection.bind(this);
-
-        this.state = {
-            selection: undefined,
-        };
-    }
+const RoomNameInput = forwardRef((props, ref) => {
```

Comment:
Hey
There's an awesome comment on migration to functional components [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1686062674061049?thread_ts=1685727880.152119&cid=C01GTK53T8Q)

> [function vs const](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#should-i-declare-my-components-with-arrow-functions-const-or-the-function-keyword) - Declare components with function keyword.